MaxText/inference_microbenchmark.py

```diff
@@ -176,6 +169,9 @@ def main(config):
   num_model_params, model_size, _ = summarize_pytree_data(params, name="Model")
 
   benchmark_results = {"Prefill": {}}
+
+  max_utils.activate_profiler(config, config.profile_name)
```

Comment:
Moving the profiling into `main` may not be ideal. 
1. It captures additional setup steps that are done as part of `ar_benchmark` and `prefill_benchmark` before running the corresponding benchmark loops. 
2. It puts all the prefill and ar kernels and setup calls into one single profile, which may or may not be a bad thing. To me it introduces more noise and makes it a little harder to clearly see the pieces we care about. But I can also see the appeal of having a single profile per run.

Reply:
I think on the balance it would be better if it was done tastefully (a.k.a. we avoid the compiles in the profile so the profiles were lean and mean).